spatialmath/box.go

```diff
@@ -352,3 +356,82 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector.
+func (b *box) ToPointCloud(resolution float64) []r3.Vector {
+	b.pose.Orientation()
+
+	// check for user defined spacing
+	var iter float64
+	if resolution != 0. {
+		iter = resolution
+	} else {
+		iter = defaultBoxPointDensity
+	}
+	var faces [][]float64
+	// create points on box faces with box centered at (0, 0, 0)
+
+	fillFaces := func(fixedDimension int) {
+		starts := [3]float64{0.0, 0.0, 0.0}
+		starts[fixedDimension] = b.halfSize[fixedDimension]
+		for i := starts[0]; i <= b.halfSize[0]; i += iter {
+			for j := starts[1]; i <= b.halfSize[1]; i += iter {
+				for k := starts[2]; k <= b.halfSize[2]; k += iter {
+					p1 := []float64{i, j, k}
+					p2 := []float64{i, j, -k}
+					p3 := []float64{i, -j, k}
+					p4 := []float64{i, -j, -k}
+					p5 := []float64{-i, j, k}
+					p6 := []float64{-i, j, -k}
+					p7 := []float64{-i, -j, -k}
+					p8 := []float64{-i, -j, k}
+
+					switch {
+					case i == 0.0 && j == 0.0:
+						faces = append(faces, p1, p2)
+					case j == 0.0 && k == 0.0:
+						faces = append(faces, p1, p5)
+					case i == 0.0 && k == 0.0:
+						faces = append(faces, p1, p7)
+					case i == 0.0:
+						faces = append(faces, p1, p2, p3, p4)
+					case j == 0.0 && fixedDimension == 0:
```

Comment:
added `&& fixedDimension == 0:` to this switch because otherwise we enter this switch statement twice